app/controllers/topics_controller.rb

```diff
@@ -0,0 +1,58 @@
+class TopicsController < ApplicationController
+  before_action :set_topic, only: %i[ show edit update destroy ]
+
+  def index
+    @topics = Topic.all
+    render json: @topics
```

Comment:
I think you don't need to call this `render` because rails automatically will pass the @topics to the view file related to this index action in /views/topics/index.json.jbuilder, or whatever that you have in the view.